src/action.ts

```diff
@@ -83,6 +84,94 @@ function pullRequestTitle(dependency: string, workingDir: string): string {
   return title
 }
 
+
+export interface PatcherUpdateSummary {
+  successful_updates:           SuccessfulUpdate[];
+  manual_steps_you_must_follow: ManualStepsYouMustFollow[];
+}
+
+export interface ManualStepsYouMustFollow {
+  instructions_file_path: string;
+}
+
+export interface SuccessfulUpdate {
+  file_path:       string;
+  updated_modules: UpdatedModule[];
+}
+
+export interface UpdatedModule {
+  repo:                  string;
+  module:                string;
+  previous_version:      string;
+  updated_version:       string;
+  next_breaking_version: NextBreakingVersion;
+  patches_applied:       PatchesApplied;
+}
+
+export interface NextBreakingVersion {
+  version:           string;
+  release_notes_url: string;
+}
+
+export interface PatchesApplied {
+  slugs: string[];
+  manual_scripts: string[];
+  count: number;
+}
+
+function pullRequestBodyUpdatedModules(modules: UpdatedModule[]): string {
+  // TODO do not show patches applied if count = 0
+  return modules.map(module => (`  - Previous version: \`${module.previous_version}\`
+  - Updated version: \`${module.updated_version}\` ([Release notes for ${module.next_breaking_version.version}](${module.next_breaking_version.release_notes_url}))
+  - Patches applied: ${module.patches_applied.count}`)).join("\n");
+}
+
+function pullRequestBodySuccessfulUpdates(updatedModules: SuccessfulUpdate[]): string {
+  if (updatedModules) {
+    return updatedModules.map(module => (`- \`${module.file_path}\`
+${pullRequestBodyUpdatedModules(module.updated_modules)}`)).join("\n");
+  }
+
+  return "";
+}
+
+function pullRequestBodyReadmeToUpdate(manualSteps: ManualStepsYouMustFollow[]): string {
+ if (manualSteps) {
```

Comment:
```suggestion
 if (manualSteps && manualSteps.length > 0) {
```

This is because the single check for truthiness of `manualSteps` returns `true` for an empty array.

Reply:
Technically speaking the first clause here is not necessary because `manualSteps` cannot be `undefined` due to the type constraint.

Reply:
Hm, maybe it's due to my local setup, but I got some weird errors with the `undefined` object 😬